narwhals/_pandas_like/dataframe.py

```diff
@@ -290,10 +284,15 @@ def join(
             right_on = [right_on]
 
         if how == "cross":
-            if self._implementation in {"modin", "cudf"} or (
-                self._implementation == "pandas"
-                and (pd := get_pandas()) is not None
-                and parse_version(pd.__version__) < parse_version("1.4.0")
+            backend = get_backend(self._implementation)
+            if (
+                self._implementation is Implementation.MODIN
+                or self._implementation is Implementation.CUDF
```

Comment:
does `self._implementation in {Implementation.MODIN, Implementation.CUDF}` work?

Reply:
Unfortunately MyPy only supports type narrowing in Enums with the `is` operator. Runtime behaviour is identical but MyPy will not narrow the type unless done this way

I believe this is because `in` uses `==` and not `is` internally but I am not totally sure